connect/api/src/main/java/org/apache/kafka/connect/rest/extension/entities/ConnectorStateDetail.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.rest.extension.entities;
+
+
+import java.util.List;
+import java.util.Objects;
+
+public class ConnectorStateDetail {
+
+    private final String name;
+    private final ConnectorState connector;
+    private final List<TaskState> tasks;
+    private final ConnectorType type;
+
+
+    public ConnectorStateDetail(String name,
+                                ConnectorState connector,
+                                List<TaskState> tasks,
+                                ConnectorType type) {
+        this.name = name;
+        this.connector = connector;
+        this.tasks = tasks;
+        this.type = type;
+    }
+
+
+    public String name() {
+        return name;
+    }
+
+
+    public ConnectorState connector() {
+        return connector;
+    }
+
+
+    public List<TaskState> tasks() {
```

Comment:
Is it intentional that these tasks are ordered? To the indexes correspond to the task ID? If not, perhaps a Collection is sufficient, or a Map keyed by task ID.